frontend_server/publish.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'package:path/path.dart' as path;
+
+const String vmPubspec = r'''
+name: vm
+version: 0.0.1
+environment:
+  sdk: ">=2.2.2 <3.0.0"
+
+dependencies:
+  front_end: any
+  kernel: any
+  meta: any
+  build_integratioon: any
+''';
+
+const String buildIntegrationPubspec = r'''
+name: build_integration
+version: 0.0.1
+environment:
+  sdk: ">=2.2.2 <3.0.0"
+
+dependencies:
+  front_end: any
+  meta: any
+''';
+
+const String frontendServerPubspec = r'''
```

Comment:
What is the reason to have `pubspec.yaml` contents hardcoded in this script?

Reply:
The pubspec of vm/build_integration are not valid, while the pubspec of front_end and kernel say not to rely on the versions. To ensure I don't have version solving issues, I've replace the pubspecs here with `any` constraints. This will be used by a generated pubspec in the flutter_tool that uses the `dependency_overrides` section to specify path versions.

Reply:
Do you know why those packages have invalid or reuse-discouraging pubspec.yamls? Is that to prevent deploying and distributing those packages for some reason? If we intend to have those packages to be distributed and available, wouldn't be cleaner to fix their pubspec.yamls instead of providing our replacements for those?

Reply:
I'm not really distributing or publishing them though, I'm just allowing them to be vendored by the flutter SDK. The versions/constraints are not really meaningful in that content, as they won't be user visible.

Reply:
Sorry what does "vendored by the flutter SDK" mean? Does this effectively gives flutter tools (and only flutter tools, because nobody else downloads flutter engine artifacts) access to those packages, while currently those packages(well, frontend server only) is only available as executable snapshot?

Reply:
Yes, the would be in the cache artifacts where flutter_tools can see the source, and use them to regenerate the snapshot to include other kernel transforms per  flutter/flutter#36738